python/ray/tune/analysis/experiment_analysis.py

```diff
@@ -52,6 +64,654 @@
 DEFAULT_FILE_TYPE = "csv"
 
 
+@PublicAPI(stability="beta")
+class NewExperimentAnalysis:
+    """Analyze results from a Ray Train/Tune experiment.
+
+    To use this class, the run must store the history of reported metrics
+    in log files (e.g., `result.json` and `progress.csv`).
+    This is the default behavior, unless default loggers are explicitly excluded
+    with the `TUNE_DISABLE_AUTO_CALLBACK_LOGGERS=1` environment variable.
+
+    Parameters:
+        experiment_checkpoint_path: Path to an `experiment_state.json` file,
+            or a directory that contains an `experiment_state.json` file.
+        default_metric: Default metric for comparing results. Can be
+            overwritten with the ``metric`` parameter in the respective
+            functions.
+        default_mode: Default mode for comparing results. Has to be one
+            of [min, max]. Can be overwritten with the ``mode`` parameter
+            in the respective functions.
+        trials: List of trials that can be accessed via `analysis.trials`.
+
+    Example:
+        TODO(justinvyu)
+    """
+
+    def __init__(
+        self,
+        experiment_checkpoint_path: Union[str, os.PathLike],
+        trials: Optional[List[Trial]] = None,
+        default_metric: Optional[str] = None,
+        default_mode: Optional[str] = None,
+        storage_filesystem: Optional[pyarrow.fs.FileSystem] = None,
+    ):
+        self.default_metric = default_metric
+        if default_mode and default_mode not in ["min", "max"]:
+            raise ValueError("`default_mode` has to be None or one of [min, max]")
+        self.default_mode = default_mode
+        if self.default_metric is None and self.default_mode is not None:
+            # If only a mode was passed, use anonymous metric
+            self.default_metric = DEFAULT_METRIC
+
+        (
+            self._fs,
+            self._experiment_fs_path,
+            self._experiment_json_fs_path,
+        ) = self._get_experiment_fs_and_path(
+            experiment_checkpoint_path, storage_filesystem
+        )
+
+        self.trials = trials or self._load_trials()
+        self._trial_dataframes = self.fetch_trial_dataframes()
```

Comment:
Should this be an internal function? I don't think users should call this manually

Reply:
(I know it's currently public, but since this is a new implementation... we can also easily add a deprecation wrapper)

Reply:
Sounds good, let's make it private.